src/components/pages/deals/Deals.js

```diff
@@ -3,31 +3,91 @@ import React, { useState, useEffect } from 'react'
 import itemsServices from '../../../services/items'
 import ProductsOnDeals from '../../products/Products'
 import SearchBar from '../../searchbar/Searchbar'
+import PaginationButton from '../../PaginationButtons/PaginationButton'
 
 import './deals.css'
 
+const PAGE_SIZE = 6
+
 const Deals = () => {
     const [deals, setDeals] = useState([])
     const [isLoading, setIsLoading] = useState(true)
+    const [noDeals, setNoDeals] = useState(false)
+    const [number_of_pages, setNumberOfPages] = useState(null)
+    const [page, setPage] = useState(1)
 
     useEffect(() => {
         const fetch_deals = async () => {
-            const query = 'isOnSale=true'
+            // Get all deals to calculate the number of pages for pagination
+            const all_deals = await itemsServices.getDeals('isOnSale=true')
+            // set number of pages
+            setNumberOfPages(Math.ceil(all_deals.items.length/PAGE_SIZE))
```

Comment:
you can use `response.total` property from the super-store api to get the total number of items and figure out the page count through that. That way you can avoid making the unneeded initial query.